components/director/internal/model/application.go

```diff
@@ -16,6 +17,14 @@ type Application struct {
 	IntegrationSystemID *string
 }
 
+func (app *Application) UpdateApplication(update ApplicationUpdateInput) {
```

Comment:
Try to rename this to a more detailed name, `UpdateApplication` is too generic. Maybe `SetFromUpdateInput`